File: src/ipc/mirror.ts
import { invoke } from "@tauri-apps/api/core";

import type { Network } from "./types";

/** Resolve a raw GitHub URL (e.g. a thumbnail) through the configured mirror */
export const mirrorUrl = async (url: string, network?: Network) =>
  invoke<string>("mirror_url", { url, network });

File: crates/dwall-settings/src/commands/mirror.rs
//! URL mirroring commands.

use dwall::config::Network;

use crate::network::MirrorUrlResolver;

/// Resolve a raw GitHub file URL (e.g. a theme thumbnail) through the configured
/// mirror template. Returns the URL unchanged when no mirror is configured.
#[tauri::command]
pub fn mirror_url(url: String, network: Option<Network>) -> String {
    MirrorUrlResolver::resolve_raw(network.as_ref(), &url)
}
